packages/smooth_app/lib/pages/product/product_image_viewer.dart

```diff
@@ -36,14 +37,19 @@ class _ProductImageViewerState extends State<ProductImageViewer> {
   late final AppLocalizations appLocalizations = AppLocalizations.of(context);
 
   /// When the image is edited, this is the new image
-  late ImageProvider imageProvider;
-  bool _isEdited = false;
+  ImageProvider? imageProvider;
+  bool isImageUrlAvailable = true;
```

Comment:
It's a bit confusing to have both fields with opposite default values.
Actually I don't think we need `bool isImageUrlAvailable` at all; it's just the same as `imageProvider != null`.

Reply:
For the record, private fields are supposed to be private. And start with `_`.

Reply:
> For the record, private fields are supposed to be private. And start with `_`.

Ohkay

Reply:
> It's a bit confusing to have both fields with opposite default values. Actually I don't think we need `bool isImageUrlAvailable` at all; it's just the same as `imageProvider != null`.

Point was to make it reusable, But the implementation complicated and quite confusing, Will make the changes, Please check it's readable enough.